frontend/src/main/components/RecommendationRequest/RecommendationRequestTable.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+
+import { useBackendMutation } from "main/utils/useBackend";
+import {
+  cellToAxiosParamsDelete,
+  onDeleteSuccess,
+} from "main/utils/RecommendationRequestUtils";
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function RecommendationRequestTable({ requests, currentUser }) {
+  const navigate = useNavigate();
+
+  const editCallback = (cell) => {
+    navigate(`/requests/edit/${cell.row.values.id}`);
+  };
+
+  // Stryker disable all : hard to test for query caching
+
+  const deleteMutation = useBackendMutation(
+    cellToAxiosParamsDelete,
+    { onSuccess: onDeleteSuccess },
+    ["/api/recommendationrequest/all"],
+  );
+  // Stryker restore all
+
+  // Stryker disable next-line all : TODO try to make a good test for this
+  const deleteCallback = async (cell) => {
+    deleteMutation.mutate(cell);
+  };
+
+  const columns = [
+    {
+      Header: "id",
+      accessor: "id", // accessor is the "key" in the data
+    },
+    {
+      Header: "Professor Name",
+      accessor: "professorName",
+    },
+    {
+      Header: "Professor Email",
+      accessor: "professorEmail",
+    },
+    {
+      Header: "Requester Name",
+      accessor: "requesterName",
+    },
+    {
+      Header: "Recommendation Type",
+      accessor: "recommendationType",
+    },
+    {
+      Header: "Details",
+      accessor: "details",
+    },
+    {
+      Header: "Status",
+      accessor: "status",
+    },
+    {
+      Header: "Submission Date",
+      accessor: "submissionDate",
+    },
+    {
+      Header: "Completion Date",
+      accessor: "completionDate",
+    },
+  ];
+
+  if (hasRole(currentUser, "ROLE_ADMIN")) {
+    columns.push(
+      ButtonColumn(
+        "Edit",
+        "primary",
+        editCallback,
+        "RecommendationRequestTable",
+      ),
+    );
```

Comment:
I don't think we want to allow admins to Edit Recommendation Requests.  I think we can allow Users to edit their OWN recommendation requests, for example (which should be the only ones that show up on that page), to update ONLY the details (and not any other field).  I think we might also want to allow users to delete their own recommendation requests, and perhaps allow Admins to delete "any" recommendation request.